src/Functions/FunctionsConversion.h

```diff
@@ -2406,6 +2407,11 @@ class CastOverloadResolver : public IFunctionOverloadResolverImpl
                 " Instead there is a column with the following structure: " + column->dumpStructure(),
                 ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
 
+        Settings set;
```

Comment:
It dows not work this way. You should get settings with `context.getSettingsRef()` that allowed in `create(const Context &)`